qiskit_experiments/library/characterization/t1.py

```diff
@@ -244,21 +266,118 @@ def parameters(self) -> np.ndarray:
         """
         opt = self.experiment_options  # alias
 
-        if opt.stark_amps is None:
+        if opt.xvals is None:
             if opt.spacing == "linear":
-                params = np.linspace(opt.min_stark_amp, opt.max_stark_amp, opt.num_stark_amps)
+                params = np.linspace(opt.min_xval, opt.max_xval, opt.num_xvals)
             elif opt.spacing == "quadratic":
-                min_sqrt = np.sign(opt.min_stark_amp) * np.sqrt(np.abs(opt.min_stark_amp))
-                max_sqrt = np.sign(opt.max_stark_amp) * np.sqrt(np.abs(opt.max_stark_amp))
-                lin_params = np.linspace(min_sqrt, max_sqrt, opt.num_stark_amps)
+                min_sqrt = np.sign(opt.min_xval) * np.sqrt(np.abs(opt.min_xval))
+                max_sqrt = np.sign(opt.max_xval) * np.sqrt(np.abs(opt.max_xval))
+                lin_params = np.linspace(min_sqrt, max_sqrt, opt.num_xvals)
                 params = np.sign(lin_params) * lin_params**2
             else:
                 raise ValueError(f"Spacing option {opt.spacing} is not valid.")
         else:
-            params = np.asarray(opt.stark_amps, dtype=float)
+            params = np.asarray(opt.xvals, dtype=float)
 
+        if opt.xval_type == "frequency":
+            return self._frequencies_to_amplitudes(params)
         return params
 
+    def _frequencies_to_amplitudes(self, params: np.ndarray) -> np.ndarray:
+        """A helper method to convert frequency values to amplitude.
+
+        Args:
+            params: Parameters representing a frequency of Stark shift.
+
+        Returns:
+            Corresponding Stark tone amplitudes.
+
+        Raises:
+            RuntimeError: When service or analysis results for Stark coefficients are not available.
+            TypeError: When attached analysis class is not valid.
+            KeyError: When stark_coefficients dictionary is provided but keys are missing.
+            ValueError: When specified Stark shift is not available.
+        """
+        opt = self.experiment_options  # alias
+
+        if not isinstance(self.analysis, StarkP1SpectAnalysis):
+            raise TypeError(
+                f"Analysis class {self.analysis.__class__.__name__} is not a subclass of "
+                "StarkP1SpectAnalysis. Use proper analysis class to scan frequencies."
+            )
+        coef_names = self.analysis.stark_coefficients_names
```

Comment:
This is okay, but I think it would be better to have a standalone class for the Stark coefficients so there is one clear place to reference for the definition of the coefficient names and the helper functions for converting amplitude to frequency. Currently, the definition is spread across this class and StarkRamseyXYAmpScanAnalysis and there could be other experiments that want to reference this data and conversion.

Reply:
This makes perfect sense. `StarkRamseyXYAmpScan` can also store (serializable) Stark coefficient class object in the analysis results instead of individual 7 entries for polynomial coefficients. This means the new class can be used by multiple analyses, and I think it's better to create a dedicated module to maintain such util class. However, this could cause conflict with #1243 , and I'll update this PR after another one is merged to keep the commit log clean.